iOSDevUK/Controllers/Maps/AllLocationsMapViewController.swift

```diff
@@ -77,11 +81,12 @@ class AllLocationsMapViewController: UIViewController, UITableViewDelegate, UITa
     func showMapLocations(forIndexPath indexPath: IndexPath) {
         mapView.removeAnnotations(mapView.annotations)
         
-        guard let locations = locationTypes?[indexPath.row].locations else {
+        guard let locationTypes = locationTypes else {
             print("Unable to access list of locations")
             return
         }
         
+        let locations = locationTypes[indexPath.row].locations
```

Comment:
The above two line changes relate to when there's no data downloaded and `locationTypes` is nil, which would then still try to subscript the nil array, which caused a crash